tests/testing_utils.py

```
@@ -117,22 +116,77 @@ def requires_gpu_mem(required_amount: Union[int, float]) -> pytest.MarkDecorator
+            else:
+                file_path = sys.modules.get(func.__module__).__file__
+                func_name = func.__name__
+
+                cmd = (
+                    f"{sys.executable} "
+                    f"-m torch.distributed.run --nproc_per_node {world_size} "
+                    "--log-dir /tmp/torchrun-logs --tee 3 --role torchrun "
+                    f"-m pytest {file_path}::{func_name} -sx"
+                )
+
+                proc = subprocess.run(cmd.split(" "))
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)